test_conformance/vulkan/test_vulkan_api_consistency.cpp

```diff
@@ -0,0 +1,496 @@
+#include <vulkan_interop_common.hpp>
+#include <opencl_vulkan_wrapper.hpp>
+#include <vulkan_wrapper.hpp>
+#include <CL/cl.h>
+#include <CL/cl_ext.h>
+#include <assert.h>
+#include <vector>
+#include <iostream>
+#include <string.h>
+#include "harness/testHarness.h"
+#include "harness/typeWrappers.h"
+#include "harness/deviceInfo.h"
+              
+int test_consistency_external_buffer(cl_device_id deviceID,
+                               cl_context _context,
+                               cl_command_queue _queue,
+                               int num_elements)
+{
+    cl_int errNum;
+    VulkanDevice vkDevice;
+    // Context and command queue creation
+    cl_platform_id platform = NULL;
+    cl_context context = NULL;
+    cl_command_queue cmd_queue = NULL;
+
+    cl_context_properties contextProperties[] =
+    {
+       CL_CONTEXT_PLATFORM,
+       0,
+       0
+    };
+    errNum = clGetPlatformIDs(1, &platform, NULL);
+    test_error(errNum, "Failed to get platform Id");
+
+    contextProperties[1] = (cl_context_properties)platform;
+
+    context = clCreateContextFromType(contextProperties, CL_DEVICE_TYPE_GPU, NULL, NULL, &errNum);
+    test_error(errNum, "Unable to create context with properties");
+
+    cmd_queue = clCreateCommandQueue(context, deviceID, 0, &errNum);
+    test_error(errNum, "Unable to create command queue");
+   
+    uint32_t bufferSize = 32;
+    cl_device_id devList[] = {deviceID, NULL};
+
+    #ifdef _WIN32
+        if (!is_extension_available(devList[0], "cl_khr_external_memory_win32")) {
+            throw std::runtime_error(" Device does not support cl_khr_external_memory_win32 extension \n");
```

Comment:
Don't we want to skip the test instead?

Reply:
I see two cases here -->
1. cl_khr_external_memory/cl_khr_external_semaphore is not supported. In which case, we should have already skipped the test. 
2. cl_khr_external_memory/cl_khr_external_semaphore is supported. In which case, not supporting any of windows specific handles, should be an error. While it doesn't need to be cl_khr_external_memory_win32 and can be cl_khr_external_memory_dx in future, the test can not be skipped unconditionally and should evaluate minimum required support before deciding on skipping or error out.